js/modules.js

```diff
@@ -15,52 +15,79 @@ CORE.create_module("email-nav", function (sb) {
 CORE.create_module("email-catalog", function (sb) {
     var emails;
 
+    var openEmail = function(e) {
+        sb.hide();
+        sb.notify({
+            type : "open-email",
+            data : e.currentTarget.getAttribute("data-email-id")
+        });
+    }
+    var selectEmailCheckbox = function(e) {
+        e.stopPropagation();
+        sb.notify({
+            type : "select-email-checkbox",
+            data : e.currentTarget.getAttribute("data-email-id")
+        });
+    }
+    var markSelectedEmailsAsRead = function() {
+        sb.notify({ 
+            type : "mark-selected-emails-as-read"
+        });
+    }
+
+    var fetchEmails = function(data) {
+        emails = data;
+        sb.template(emails);
+    }
+
     return {
         init : function() {
-            sb.onEvent(".email-catalog_email", "click", this.openEmail.bind(this));
+            sb.onEvent(".email-catalog_email", "click", openEmail);
+            sb.onEvent(".email-catalog_email_checkbox", "click", selectEmailCheckbox);
+            sb.onEvent("button", "click", markSelectedEmailsAsRead);
```

Comment:
What you mean? we need the click handler to work on each email which each have the `.email-catalog_email` class. The onEvent is using the $.on() behind the scenes for event delgation. It uses the module name as the parent `id` that is putting the events on